### DISCUSSION TITLE
How to use drop in Development

### DISCUSSION THREAD

**Discussion by pdavidsonFIA** [casact/chainladder-python#166] [Q&A]
What's a convenient way to drop an entire origin from a development?
Currently dropping from the raw data before loading to a triangle, but this can cause some bugs.

**Comment by jbogaardt** [comment]
This is a good question as its not straight forward.

### Dropping just in the triangle, without Development estimator
This **doesn't work**, because broadcasting takes priority over label matching:
```python
import chainladder as cl
raa = cl.load_sample('raa')
# Exclude only 1984  - Fail
raa - raa[raa.origin=='1984'] 
```
But if I eliminate more than one origin and bring back the ones I want to keep, then label matching takes priority. **This works:**
```python
# Exclude only 1984 - Success
raa - raa[raa.origin>'1984'] + raa[raa.origin>'1985']
```
This all assumes you want to retain the origin period, just have it zero/blank. If you want to completely eliminate even the placeholder for the origin period, then you can do this:
```python
# Exclude only 1984 - Success with consequences? 
raa[raa.origin<'1984'] + raa[raa.origin>'1985']
```
However, the last one will almost certainly make any of the estimators of the package fail.


### Dropping in the development estimator itself:
There is no `drop_origin` in the Development estimator, but it can be emulated with the `drop` parameter by listing each individual cell of the origin you want to eliminate.  For example:

```python
import chainladder as cl
raa = cl.load_sample('raa')
cl.Development(drop=list(zip(['1984']*6, range(12, 84, 12)))).fit(raa).ldf_
```

Final thought - I fully expect that dropping the origin before creating the triangle should produce identical results.  I am bothered by the fact that this is not the case for you.

**Reply by mlbugs** [reply]
Am working on something similar to the last point, only with trying to drop all but the latest few valuation periods in my triangle.  I can go back to the triangle itself, drop all prior valuations periods and then fit ldf's on the result.  Having 'keep' options within the development estimator would make life easier so I don't have to go back to the original triangle to do the manipulations on the triangle.  Or I can learn how to do fancy stuff like this: ` list(zip(['1984']*6, range(12, 84, 12)))`

**Reply by mlbugs** [reply]
Doh! I should just be using n_periods.  At least I learned to do it a harder way first.
